server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/graphql/GraphQLLoggerUtil.java

```diff
@@ -63,43 +65,97 @@ public static WebSession getWebSession(HttpServletRequest request) {
-        StringBuilder loggerMessage = new StringBuilder(" [user: ").append(userId)
-            .append(", sessionId: ").append(sessionId).append("]");
-
-        if (WebAppUtils.getWebPlatform().getPreferenceStore().getBoolean(LOG_API_GRAPHQL_DEBUG_PARAMETER)
-                && variables != null
-        ) {
-            loggerMessage.append(" [variables] ");
-            String parsedVariables = parseVarialbes(variables);
-            if (CommonUtils.isNotEmpty(parsedVariables)) {
-                loggerMessage.append(parseVarialbes(variables));
+    public static String buildLoggerMessage(String sessionId, String userId, Method method, Object[] args) {
+        StringBuilder sb = new StringBuilder(64)
+            .append(" [user: ").append(userId)
+            .append(", sessionId: ").append(sessionId)
+            .append("]");
+
+        if (WebAppUtils.getWebPlatform().getPreferenceStore().getBoolean(LOG_API_GRAPHQL_DEBUG_PARAMETER)) {
+            sb.append('(');
+            String text = maskArgsToString(method, args);
+            if (CommonUtils.isNotEmpty(text)) {
+                sb.append(text);
             }
+            sb.append(')');
         }
-        return loggerMessage.toString();
+        return sb.toString();
     }
 
-    private static String parseVarialbes(Map<String, Object> map) {
-        StringBuilder result = new StringBuilder();
-
-        for (Map.Entry<String, Object> entry : map.entrySet()) {
-            String key = entry.getKey();
-            Object value = entry.getValue();
+    public static String maskArgsToString(Method method, Object[] args) {
+        Parameter[] params = method.getParameters();
+        if (params.length == 0 || args == null || args.length == 0) {
+            return "";
+        }
 
-            boolean isProhibited = PROHIBITED_VARIABLES.stream()
-                .anyMatch(prohibitedKey -> key.toLowerCase().contains(prohibitedKey.toLowerCase()));
+        int limit = Math.min(args.length, params.length);
+        StringJoiner joiner = new StringJoiner(", ");
 
-            if (isProhibited) {
-                result.append(key).append(": ").append("******** ");
+        for (int i = 0; i < limit; i++) {
+            Object value = args[i];
+            if (value instanceof WebSession) {
+                //we already log sessionId
+                continue;
+            }
+            if (params[i].isAnnotationPresent(WebParameterSecure.class)) {
+                joiner.add("****");
+                continue;
+            }
+            if (value instanceof String sv && CommonUtils.isEmpty(sv)) {
                 continue;
             }
 
-            if (value instanceof Map) {
-                result.append(parseVarialbes((Map<String, Object>) value));
+            if (value != null && !isSimple(value.getClass())) {
+                StringJoiner fields = new StringJoiner(", ");
```

Comment:
Why so complicated? Let's just serialize it to json with GSON. Or even use toString.